sympy/functions/elementary/exponential.py

```diff
@@ -436,7 +436,11 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n + 2
+        cf = Order(arg.as_leading_term(x), x).getn()
+        if cf != 0:
+            nterms = (n/cf).ceiling()
+        exp_series = exp(t)._taylor(t, n, nterms)
```

Comment:
So passing nterms for n is not the right thing to do?

Reply:
I am not replacing the n.I have added the extra parameter `nterms` so that we can calculate the no of terms accurately.(that is working) what we are struggling here is to get the right order term.

Please correct me If I am saying or doing something wrong.

Reply:
`n` is only needed for the order term `O(x**n)`. It is not necessary to add that in `_taylor`. Besides, it stripped on line 448.

Reply:
It is also used on line 469.
`g = g.nseries(x, n=n)`

Reply:
I mean that it is not necessary to pass `n` to `_taylor`. It can return a truncated series with no order term. Only the last line need be changed (and the call parameter in `_taylor(t, nterms)`).